doc/nextml2015/presentation.tex

```diff
@@ -0,0 +1,1186 @@
+  \end{itemize}
+\end{frame}
+
+
+
+%% \begin{frame}{Why scripting for GPUs?}
+%%   \begin{bf}They complement each other\end{bf}
+
+%%   GPUs are everything that high level languages are not
+
+%%   \begin{itemize}
+%%     \item Highly parallel
+%%     \item Very architecture-sensitive
+%%     \item Built for maximum FP/memory throughput
+%%     \item So hard to program that meta-programming is easier
+%%   \end{itemize}
+
+%%   \begin{bf}Best of both worlds:\end{bf} easily scripted code which invokes high-performance GPU kernels.
+
+%%   \begin{bf}Theano C code generation removes overhead\end{bf} of
+%%   function calls between Python and C by launching many C functions at once.
+
+%% \end{frame}
+
+\begin{frame}{Project status?}
+  \begin{itemize}
+    \item Mature: Theano has been developed and used since January 2008 (7 yrs old)
+    \item Driven hundreds research papers
+    \item Good user documentation
+    \item Active mailing list with participants from outside our lab
+    \item Core technology for a few Silicon-Valley start-ups
+    \item Many contributors (some from outside our lab)
+    \item Used to teach many university classes
+    \item Has been used for research at big compagnies
+  \end{itemize}
+  Theano: \url{deeplearning.net/software/theano/}
+
+  Deep Learning Tutorials: \url{deeplearning.net/tutorial/}
+\end{frame}
+
+
+%% \begin{frame}{Overview of Library}
+%%   Theano is many things
+%%   \begin{itemize}
+%%   \item Language
+%%   \item Compiler
+%%   \item Python library
+%%   \end{itemize}
+%% \end{frame}
+
+\begin{frame}{Overview Language}
+  \begin{itemize}
+  \item Operations on scalar, vector, matrix, tensor, and sparse variables
+  \item Linear algebra
+  \item Element-wise nonlinearities
+  \item Convolution
+  \item Indexing, slicing and advanced indexing.
+  \item Reduction
+  \item Dimshuffle (n-dim transpose)
+  \item Extensible
+  \end{itemize}
+\end{frame}
+
+
+\begin{frame}[fragile]
+  \frametitle{Overview}
+  Using Theano:
+  \begin{itemize}
+  \item define expression $f(x,y) = x + y$
+  \item compile expression
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        }
+\begin{lstlisting}
+int f(int x, int y){
+  return x + y;
+}
+\end{lstlisting}
+
+  \item execute expression
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        }
+\begin{lstlisting}
+>>> f(1, 2)
+3
+\end{lstlisting}
+  \end{itemize}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{Scalar math}
+  Using Theano:
+  \begin{itemize}
+  \item define SYMBOLIC expression $f(x,y) = x + y$
+  \end{itemize}
+\lstset{language=Python,
+        commentstyle=\itshape\color{blue},
+        stringstyle=\color{violet},
+        }
+\begin{lstlisting}
+import theano
+from theano import tensor as T
+x = T.scalar()
+y = T.scalar()
+z = x+y
+w = z*x
+a = T.sqrt(w)
+b = T.exp(a)
+c = a ** b
+d = T.log(c)
```

Comment:
I updated the PR. Can you check again? In particular the RNN section.

thanks

On Thu, Jan 15, 2015 at 10:47 AM, abergeron <notifications@github.com>
wrote:

> In doc/nextml2015/presentation.tex
> <https://github.com/Theano/Theano/pull/2405#discussion-diff-23032764>:
>
> > +  \end{itemize}
> > +\lstset{language=Python,
> > +        commentstyle=\itshape\color{blue},
> > +        stringstyle=\color{violet},
> > +        }
> > +\begin{lstlisting}
> > +import theano
> > +from theano import tensor as T
> > +x = T.scalar()
> > +y = T.scalar()
> > +z = x+y
> > +w = z*x
> > +a = T.sqrt(w)
> > +b = T.exp(a)
> > +c = a ** b
> > +d = T.log(c)
>
> I think these extra expressions on this slide would just confuse people.
> They did confuse me for a while as I was pondering their meaning. You
> should either remove them or adjust the description above. Otherwise it can
> be taken as "doing an addition in theano is really hard: look at all the
> code you have to write which doesn't even seem to be related but must be
> necessary since it's on the slide."
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/2405/files#r23032764>.
>